src/client.js

```diff
@@ -1,24 +1,53 @@
 /* eslint-env browser */
 import React from 'react';
+
+// Provider makes the store available to all connect() calls in the
+// component hierarchy below. (connect() is what connects a React
+// component to a Redux store.)
 import { Provider } from 'react-redux';
+
+// ReactDOM is part of isomorphic React. It renders to HTML or the DOM.
+// ex. ReactDOM.render(<MyComponent />, node)
 import ReactDOM from 'react-dom';
+
+// fromJS will convert nested Objects and Arrays to Immutable.Map and
+// Immutable.List, respectively
 import { fromJS } from 'immutable';
+
+// Router keeps the URL in sync with what's being displayed on the page
 import { Router } from 'react-router';
+
+// history abstracts away the differences in platforms and provides a
+// minimal API to manage history. we are only including the web
+// module
 import createBrowserHistory from 'history/lib/createBrowserHistory';
+
+// pull in src/routes.js into the variable "getRoutes". It is a function
+// that takes one variable, which is the Redux store.
 import getRoutes from 'routes';
+
+// developer sidebar React component
 import DevTools from './containers/DevTools/DevTools';
 
+// polyfill the global environment (either in Node or in the browser via CommonJS)
 require('es6-promise').polyfill();
+
+// adds fetch() as a global so that its API is consistent between client and server.
 require('isomorphic-fetch');
 
+// __INITIAL_STATE__ is a JS object inlined with React's dangerouslySetInnerHTML
```

Comment:
`__INITIAL_STATE__` allows the server to tell the client how it rendered the components.